server/src/main/java/io/unitycatalog/server/service/IcebergRestCatalogService.java

```diff
@@ -44,6 +43,7 @@ public IcebergRestCatalogService(
     this.catalogService = catalogService;
     this.schemaService = schemaService;
     this.tableService = tableService;
+    this.metadataService = new MetadataService(new FileIOFactory());
```

Comment:
I feel it's better to make either the MetadataService or the FileIOFactory depdency-injectable so that we don't need to spy on them in unit tests, especially for classes that has side effect such as performing I/O.

Reply:
I mean I can pass a metadata service into this constructor if you'd like, but there really isn't any dependency injection happening in this project as it is (and nothing else relies on metadata service currently).

I also don't really follow the argument on not spy'ing/mock'ing, are you saying we should write a whole MockFileIOFactory class just to test?

Reply:
Let me rephrase. I want to minimize the use of spy() because it's often unclear which part of the implementation is mocked and which part is tested. But mocking is fine. We don't need to write a separate class just for testing.

In `MetadataServiceTest`, can we just mock `FileIOFactory.getFileIO` so that it returns different result for each test case? We should have a separate unit test for `FileIOFactory.getFileIO` if needed.

Reply:
that's fine, I switched from spy to mock